### DISCUSSION TITLE
Get Old Install Path - Using Custom Action

### DISCUSSION THREAD

**Discussion by BartoszGoralski** [wixtoolset/issues#7653] [Questions] [answered]
Hello,

For our product, we need to get old installation path (in case of installin new wersion of product with shared Upgrade Code).
We don't save InstallDir to registry Entry, so we need to use custom dll with instructions. I tried a lot of examples and cases...

Behavior shoult be like this:
If older version of product is installed -> get old install dir (changed by user propably) -> save it in to property INSTALLDIR -> pass property to 		`<Property Id="WIXUI_INSTALLDIR" Value="INSTALLDIR" />` and then continue installation in old folder.

Any ideas how to do it? Or is there any other solution than custom dll?

**Comment by robmen** [comment, marked as answer]
I'd try to use a [ComponentSearch](https://wixtoolset.org/docs/schema/wxs/componentsearch/).
